### PR TITLE
enhancement: manifest asset mapping, ACI status buttons in  list view, updated ame…

### PR DESCRIPTION
**Pull request by Hardeep8587**
…nd object in both manifests, sonar issues in fuel,alert,ifta

## Change Description

1. cross border -> asset mapping (both manifests) -> design and development
2. sonar issues-> fuel,alert and IFTA
3. ACI list view - > status buttons background
4. Amend object updated in both manifests( cross border) -> created a function to optimize code

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist: (Check only applicable)

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Hardeep8587** [comment]
@kunalfleethawks  please review and merge ! thanku!

File: src/app/pages/compliance/ifta/mileage/mileage.component.ts [hunk 1]
```diff
@@ -61,16 +60,16 @@ export class MileageComponent implements OnInit {
...
-      baseState : this.baseState,
-      baseCountry: this.baseCountry,
-      accountNumber:  this.accountNumber,
-      signingAuthority: {
-        name: '',
-        phone: '',
-        title: '',
-      }
-    };
+    // const data = {
```
Comment (kunalfleethawks) [inline src/app/pages/compliance/ifta/mileage/mileage.component.ts:63]: Is this method required?

File: src/app/pages/dispatch/cross-border/ace-documents/new-ace-manifest/new-ace-manifest.component.ts [hunk 2]
```diff
@@ -190,7 +192,7 @@ export class NewAceManifestComponent implements OnInit {
       this.modalTitle = 'Edit';
       this.fetchACEEntry();
       this.route.queryParams.subscribe((params) => {
-        if(params.amendManifest != undefined){
+        if(params.amendManifest !== undefined){
```
Comment (kunalfleethawks) [inline src/app/pages/dispatch/cross-border/ace-documents/new-ace-manifest/new-ace-manifest.component.ts:195]: What will happen if params are null?

File: src/app/pages/dispatch/cross-border/ace-documents/new-ace-manifest/new-ace-manifest.component.ts [hunk 3]
```diff
@@ -262,8 +264,18 @@ export class NewAceManifestComponent implements OnInit {
...
+      console.log('assets', this.assets);
     });
   }
+  /***
+   * fetch asset types from mapped table
+   */
+ async getBorderAssetTypes(e) {
+    const assetID = e;
+    let fetchedAsset = await this.apiService.getData('assets/' + assetID).toPromise();
+    let resultData = await this.apiService.getData('borderAssetTypes/' +   fetchedAsset.Items[0].assetDetails.assetType).toPromise(); // border aset types are fetched whose parent is asset type of selected asset
```
Comment (kunalfleethawks) [inline src/app/pages/dispatch/cross-border/ace-documents/new-ace-manifest/new-ace-manifest.component.ts]: What will happen if fetchedAsses it null and fetchedAssets.count is 0? Will it fail? Can you add defensive code to handle it?

File: src/app/pages/dispatch/cross-border/aci-documents/new-aci-manifest/new-aci-manifest.component.ts [hunk 4]
```diff
@@ -326,7 +328,15 @@ export class NewAciManifestComponent implements OnInit {
...
     });
   }
-
+ /***
+   * fetch asset types from mapped table
+   */
+  async getBorderAssetTypes(e) {
+    const assetID = e;
+    let fetchedAsset = await this.apiService.getData('assets/' + assetID).toPromise();
+    let resultData = await this.apiService.getData('borderAssetTypes/' +   fetchedAsset.Items[0].assetDetails.assetType).toPromise(); // border aset types are fetched whose parent is asset type of selected asset
```
Comment (kunalfleethawks) [inline src/app/pages/dispatch/cross-border/aci-documents/new-aci-manifest/new-aci-manifest.component.ts:337]: Same comment about fetchdAsset

**Review by kunalfleethawks — CHANGES_REQUESTED** [review]

File: src/app/pages/dispatch/cross-border/ace-documents/new-ace-manifest/new-ace-manifest.component.ts (hunk 3 above)
Comment (Hardeep8587) [reply src/app/pages/dispatch/cross-border/ace-documents/new-ace-manifest/new-ace-manifest.component.ts]: sure

File: src/app/pages/dispatch/cross-border/aci-documents/new-aci-manifest/new-aci-manifest.component.ts (hunk 4 above)
Comment (Hardeep8587) [reply src/app/pages/dispatch/cross-border/aci-documents/new-aci-manifest/new-aci-manifest.component.ts:337]: sure

File: src/app/pages/compliance/ifta/mileage/mileage.component.ts (hunk 1 above)
Comment (Hardeep8587) [reply src/app/pages/compliance/ifta/mileage/mileage.component.ts:63]: yes, but not now

File: src/app/pages/dispatch/cross-border/ace-documents/new-ace-manifest/new-ace-manifest.component.ts (hunk 2 above)
Comment (Hardeep8587) [reply src/app/pages/dispatch/cross-border/ace-documents/new-ace-manifest/new-ace-manifest.component.ts:195]: it is true and false

File: src/app/pages/dispatch/cross-border/aci-documents/new-aci-manifest/new-aci-manifest.component.ts (hunk 4 above)
Comment (Hardeep8587) [reply src/app/pages/dispatch/cross-border/aci-documents/new-aci-manifest/new-aci-manifest.component.ts:337]: I will resolve and push again

**Comment by Hardeep8587** [comment]
@kunalfleethawks please review I updated same branch with defensive code, and assets api is on change function so getting id on change

File: src/app/pages/dispatch/cross-border/aci-documents/new-aci-manifest/new-aci-manifest.component.ts (hunk 4 above)
Comment (Hardeep8587) [reply src/app/pages/dispatch/cross-border/aci-documents/new-aci-manifest/new-aci-manifest.component.ts:337]: done

**Pull request merged**